freedomofpress/securedrop
# Speed up debian build process

## Status

Ready for Review

## Description of Changes

Fixes #2939 . 
Partially addresses #3237  - moves all apt dependencies into docker for caching

Changes proposed in this pull request:

This PR parallelizes the debian build logic and moves all the apt packaging logic into docker (and out of ansible). 

## Testing

How should the reviewer test this PR?

* Run `time make build-debs` - confirm you receive no errors. The timing here will be a first run when docker has to build the image completely.
* Run `time make build-debs` again. This time the docker image doesnt have to get rebuilt and you should see a dramatic time change.
* To compare with existing switch back to `develop` and run `time make build-debs`

If you want to clean out all these docker images first you can run the following selective nuke command `docker image prune -a -f --filter "label=image_name=trusty-sd-builder-app"`

## Deployment

Any special considerations for deployment?

This code re-mixes the build process a bit and removes some tests. Give careful review and feedback to that when reviewing this PR.

We want to make sure that there are no regression in the build process so I would say that during QA we should be looking out for any weird behavior during install or involving missing files.
